File: app/src/main/java/com/pr0gramm/app/ui/intro/GenericAppIntroFragment.java
package com.pr0gramm.app.ui.intro;

import android.os.Bundle;

import com.github.paolorotolo.appintro.AppIntroBaseFragment;
import com.pr0gramm.app.R;

public final class GenericAppIntroFragment extends AppIntroBaseFragment {
    public static GenericAppIntroFragment newInstance(CharSequence title, CharSequence description, int imageDrawable, int bgColor) {
        return newInstance(title, description, imageDrawable, bgColor, 0, 0);
    }

    public static GenericAppIntroFragment newInstance(CharSequence title, CharSequence description, int imageDrawable, int bgColor, int titleColor, int descColor) {
        GenericAppIntroFragment slide = new GenericAppIntroFragment();
        Bundle args = new Bundle();
        args.putString(ARG_TITLE, title.toString());
        args.putString(ARG_DESC, description.toString());
        args.putInt(ARG_DRAWABLE, imageDrawable);
        args.putInt(ARG_BG_COLOR, bgColor);
        args.putInt(ARG_TITLE_COLOR, titleColor);
        args.putInt(ARG_DESC_COLOR, descColor);
        slide.setArguments(args);
        return slide;
    }

    @Override
    protected int getLayoutId() {
        return R.layout.intro_fragment_generic;
    }
}

File: app/src/main/java/com/pr0gramm/app/ui/intro/GenericAppIntroFragment.kt
package com.pr0gramm.app.ui.intro

import android.os.Bundle
import com.github.paolorotolo.appintro.AppIntroBaseFragment
import com.pr0gramm.app.R

class GenericAppIntroFragment : AppIntroBaseFragment() {
    override fun getLayoutId(): Int {
        return R.layout.intro_fragment_generic
    }

    companion object {
        fun newInstance(title: CharSequence, description: CharSequence, imageDrawable: Int, bgColor: Int,
                        titleColor: Int = 0, descColor: Int = 0): GenericAppIntroFragment {

            val slide = GenericAppIntroFragment()
            slide.arguments = Bundle().apply {
                putString(AppIntroBaseFragment.ARG_TITLE, title.toString())
                putString(AppIntroBaseFragment.ARG_DESC, description.toString())
                putInt(AppIntroBaseFragment.ARG_DRAWABLE, imageDrawable)
                putInt(AppIntroBaseFragment.ARG_BG_COLOR, bgColor)
                putInt(AppIntroBaseFragment.ARG_TITLE_COLOR, titleColor)
                putInt(AppIntroBaseFragment.ARG_DESC_COLOR, descColor)
            }

            return slide
        }
    }
}
